File: kyw0716/백준/브루트_포스_알고리즘/토너먼트.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const afterRound = (player) => {
  if (player % 2 === 0) {
    return player / 2;
  }
  return (player + 1) / 2;
};

rl.on("line", (input) => {
  let [round, kim, lim] = input.split(" ").map(Number);
  let count = 1;

  while (Math.abs(kim - lim) !== 1 || Math.max(kim, lim) % 2 !== 0) {
    kim = afterRound(kim);
    lim = afterRound(lim);
    count += 1;
  }

  console.log(count);
});

File: sk000801/백준/브루트포스/토너먼트.java
import java.util.*;

public class 토너먼트 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        
        int n = in.nextInt();
        int kim = in.nextInt();
        int lim = in.nextInt();
        int count = 0;

        while(kim != lim) {
            kim = (kim+1)/2;
            lim = (lim+1)/2;
            count ++;
            n /= 2;        
        }

        System.out.println(count);
    }
}

//다음 경기의 순서는 (원래수+1)/2
// 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16
//  1   2   3   4   5     6     7     8
//    1       2        3           4
//        1                 2
//                 1
